lib/inspec/rule.rb

```diff
@@ -104,7 +105,15 @@ def source_file
     # @return [nil]
     def only_if
       return unless block_given?
-      @__skip_rule ||= !yield
+
+      # Ignore error if a mock connection
+      # Example: `inspec check` with `only_if { os.name.include?('windows' }`
+      begin
+        @__skip_rule ||= !yield
+      rescue => e
+        return if @__mock == true
+        raise e
```

Comment:
You can eliminate the `e` here - `raise` within a `rescue` will re-raise the original exception.